### DISCUSSION TITLE
Split & Joins - condition feature

### DISCUSSION THREAD

**Discussion by chrisw14** [imixs/imixs-workflow#729] [Q&A]
Hi,
I'm using the Split & Joins plugin and want to ask if it's possible to start a new subprocess with the plugin if and only if a specific condition (a value of a workitem is set) is fulfilled?
Thanks!
Chris

**Comment by rsoika** [comment]
Yes - you first create a conditional event where you decide the follow up event. One contains the Create-Split instruction, the other one dose nothing.

**Comment by chrisw14** [comment]
Wouldn't it be nice to introduce a condition-tag in the workflow result?
```
<item name="subprocess_create">
    <condition>workitem.startSubProcess[0] == true</condition>
    ...
</item>
```

Because if there are more subprocesses to start there could be lots of combinations to check.

But the nicest way would be to introduce the bpmn element "Sub Process" that you can combine the conditional event with the subprocess element. Why doesn't imixs support that?
Or is it possible to change the icon of the imixs workflow event? Maybe this would be an easier solution.

**Comment by rsoika** [comment]
The idea with the condition is good. We need to look into this as a new feature. 
The bpmn element "Sub Process" does not convince me. It hides more than it makes a model transparent.

**Comment by rsoika** [comment]
This is how we typically design a process with a sub process:

![image](https://user-images.githubusercontent.com/1339124/40203571-e9ffc1a4-5a25-11e8-8c7d-75a22c01facf.png)

**Comment by rsoika** [comment]
This is the corresponding model file:
https://github.com/imixs/imixs-workflow/blob/master/imixs-workflow-core/src/test/resources/bpmn/collaboration_messageflow_complex.bpmn

**Comment by rsoika** [comment]
![image](https://user-images.githubusercontent.com/1339124/40204477-fed589f8-5a28-11e8-9fe2-6a776c31a751.png)

**Comment by rsoika** [comment]
Just to separate things:
* _Conditional-Events_ are just evaluating a rule and did not create a new instance
* _Split-Events_ are creating a new process instance based on the current process instance within the same model

So both events are running in the same model and same model-group

In difference the _SplitAndJoinPlugin_ is used to create a new process instance (named sub process) which can be part of a complete different model. And for that reason we expect a $modelversion, a $processid and a $activityid

```
<item name="subprocess_create">
    <modelversion>1.0.0</modelversion>
    <processid>100</processid>
    <activityid>10</activityid>
</item>
```

As we can place such an event after a conditional event we already have a working solution I think?

**Comment by rsoika** [comment]
It would look like this:

![image](https://user-images.githubusercontent.com/1339124/40204959-91ac0bfc-5a2a-11e8-8fdb-6d4920ed0fa4.png)

So I think we do not need the condition in a 'subprocess_create' definition ?

**Comment by chrisw14** [comment]
> As we can place such an event after a conditional event we already have a working solution I think?

Yes I thought about that yesterday too and of course this is a good solution too.

Would it be possible in that case to change the icon of the imixs-workflow-event from the yellow circle to a sub process icon?
